zkevm-circuits/src/super_circuit.rs

```diff
@@ -128,20 +130,29 @@ impl<F: Field> SubCircuitConfig<F> for SuperCircuitConfig<F> {
         let u16_table = UXTable::construct(meta);
 
         // Use a mock randomness instead of the randomness derived from the challenge
-        // (either from mock or real prover) to help debugging assignments.
+        // (either from mock or real prover) to help debugging assignments, when "mock-challenge"
+        // feature is enabled.
+        #[allow(unused_variables)]
         let power_of_randomness: [Expression<F>; 31] =
             array::from_fn(|i| Expression::Constant(mock_randomness.pow([1 + i as u64, 0, 0, 0])));
 
-        let challenges = Challenges::mock(
+        // Use the real challenges for real use, when "mock-challenge" feature is disabled.
+        #[cfg(not(feature = "mock-challenge"))]
+        let challenges = Challenges::construct(meta);
+
+        #[cfg(feature = "mock-challenge")]
+        let challenges_exprs = Challenges::mock(
             power_of_randomness[0].clone(),
             power_of_randomness[0].clone(),
         );
+        #[cfg(not(feature = "mock-challenge"))]
+        let challenges_exprs = { challenges.exprs(meta) };
```

Comment:
nitpick: I think we don't need `{ }`

Reply:
Thanks!  Fixed in https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1763/commits/4fe16a01d0f6cad3199633038a45d27cc118253c